### ISSUE TITLE
False positive in `implicit_return` and `#if`

### ISSUE THREAD

**Issue by hiragram** [realm/SwiftLint#3261]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
  - 0.39.2
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When the code is included in preprocessor `#if` statement, false positive `implicit_return` violations are reported.
This happens with properties, but not closures and functions.
When I remove `#if ...` and `#endif`, no violations will be reported.

An example is put at the end of description.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --no-cache
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'implicit_return.swift' (1/1)
implicit_return.swift:6:11: warning: Implicit Return Violation: Prefer implicit returns in closures, functions and getters. (implicit_return)
implicit_return.swift:8:11: warning: Implicit Return Violation: Prefer implicit returns in closures, functions and getters. (implicit_return)
Done linting! Found 2 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
  - 0.39.2

* Installation method used (Homebrew, CocoaPods, building from source, etc)?
  - homebrew

* Paste your configuration file:

```yml
opt_in_rules:
  - implicit_return
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
  - no

* Which Xcode version are you using (check `xcodebuild -version`)?

```bash
$ xcodebuild -version
Xcode 11.5
Build version 11E608c
```

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
class SomeClass {
    #if SOME_COMPILER_FLAG

    var someProperty: Int {
      if someFlag {
        return 1 // Implicit Return Violation
      } else {
        return 2 // Implicit Return Violation
      }
    }

    var someClosure: () -> Int = {
      if someFlag {
        return 1 // ok
      } else {
        return 2 // ok
      }
    }

    func someFunction() -> Int {
      if someFlag {
        return 1 // ok
      } else {
        return 2 // ok
      }
    }

    #endif
}
```

**Comment by rmaloney3** [comment]
I'm running into this false-positive as well with the following example as well:
```swift
#if DEBUG
private var randomBool: Bool {
    guard .random() else {
        return false // Implicit Return Violation
    }

    return true
}
#endif
```

**Comment by marcelofabri** [comment]
I think this might be a bug in SourceKit. 

- Response with `#if`: 

```json
{
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 149,
  "key.offset" : 0,
  "key.substructure" : [
    {
      "key.accessibility" : "source.lang.swift.accessibility.private",
      "key.attributes" : [
        {
          "key.attribute" : "source.decl.attribute.private",
          "key.length" : 7,
          "key.offset" : 10
        }
      ],
      "key.bodylength" : 101,
      "key.bodyoffset" : 40,
      "key.kind" : "source.lang.swift.decl.var.global",
      "key.length" : 124,
      "key.name" : "randomBool",
      "key.namelength" : 10,
      "key.nameoffset" : 22,
      "key.offset" : 18,
      "key.typename" : "Bool"
    }
  ]
}
```

- Response without `#if`:
```json
{
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 132,
  "key.offset" : 0,
  "key.substructure" : [
    {
      "key.accessibility" : "source.lang.swift.accessibility.private",
      "key.attributes" : [
        {
          "key.attribute" : "source.decl.attribute.private",
          "key.length" : 7,
          "key.offset" : 0
        }
      ],
      "key.bodylength" : 101,
      "key.bodyoffset" : 30,
      "key.kind" : "source.lang.swift.decl.var.global",
      "key.length" : 124,
      "key.name" : "randomBool",
      "key.namelength" : 10,
      "key.nameoffset" : 12,
      "key.offset" : 8,
      "key.typename" : "Bool"
    },
    {
      "key.elements" : [
        {
          "key.kind" : "source.lang.swift.structure.elem.condition_expr",
          "key.length" : 9,
          "key.offset" : 41
        }
      ],
      "key.kind" : "source.lang.swift.stmt.guard",
      "key.length" : 78,
      "key.namelength" : 0,
      "key.nameoffset" : 0,
      "key.offset" : 35,
      "key.substructure" : [
        {
          "key.bodylength" : 0,
          "key.bodyoffset" : 49,
          "key.kind" : "source.lang.swift.expr.tuple",
          "key.length" : 2,
          "key.namelength" : 0,
          "key.nameoffset" : 0,
          "key.offset" : 48
        },
        {
          "key.bodylength" : 55,
          "key.bodyoffset" : 57,
          "key.kind" : "source.lang.swift.stmt.brace",
          "key.length" : 57,
          "key.namelength" : 0,
          "key.nameoffset" : 0,
          "key.offset" : 56
        }
      ]
    }
  ]
}
```

However, this doesn't happen for functions. I recommend filling a bug on https://bugs.swift.org/ and posting the ticket link here.

**Comment by kacper1703** [comment]
Also happens in SDK 0.41.0
```swift
    #if CONDITION
    private var someValue: Int? {
        guard let integer = Int("3") else { ↓return nil }
        return integer
    }
    #endif
```
Preprocessor check removal "fixes" the issue.

**Comment by mildm8nnered** [comment]
Still present in 0.50.3

**Comment by marcelofabri** [comment]
This was fixed by https://github.com/realm/SwiftLint/pull/5166
